test/lit/passes/translate-to-new-eh.wast

```diff
@@ -1788,13 +1788,15 @@
   )
 
   ;; CHECK:      (func $deletate-target-outer-try-unreachable (type $1)
-  ;; CHECK-NEXT:  (try_table
-  ;; CHECK-NEXT:   (throw_ref
-  ;; CHECK-NEXT:    (block $l00 (result exnref)
-  ;; CHECK-NEXT:     (try_table (catch_all_ref $l00)
-  ;; CHECK-NEXT:      (call $foo)
+  ;; CHECK-NEXT:  (block $outer1
```

Comment:
Why was this outer block needed?

Reply:
Before we go into actually processing `catch`es and `delegate`s, we create an outer block in this condition:
https://github.com/WebAssembly/binaryen/blob/219e668e87b012c0634043ed702534b8be31231f/src/passes/TranslateEH.cpp#L677
Because
1. In case of a `delegate`, it will be translated into a `throw_ref` to the delegate target. But when an exception does not occur, we have to bail out of that `throw_ref`. And this outer block is that bail-out destination. It is explained here:
https://github.com/WebAssembly/binaryen/blob/e50a2ba8e85c274880898ef7c5d6efc186efcc17/src/passes/TranslateEH.cpp#L251
https://github.com/WebAssembly/binaryen/blob/e50a2ba8e85c274880898ef7c5d6efc186efcc17/src/passes/TranslateEH.cpp#L267

2. In case of `catch`es, we proceed into catch handler parts unless we bail out in case an exception does not occur. It is explained here:
https://github.com/WebAssembly/binaryen/blob/e50a2ba8e85c274880898ef7c5d6efc186efcc17/src/passes/TranslateEH.cpp#L415
https://github.com/WebAssembly/binaryen/blob/e50a2ba8e85c274880898ef7c5d6efc186efcc17/src/passes/TranslateEH.cpp#L436

But this test does not need a `br` because the body's type is unreachable due to a `return`. We omit that `br` here:
https://github.com/WebAssembly/binaryen/blob/e50a2ba8e85c274880898ef7c5d6efc186efcc17/src/passes/TranslateEH.cpp#L298-L300

It is hard to check this condition before creating an outer block. We can run `BranchSeeker` to find out if the block is actually used as a destination for a branch afterwards to remove this unnecessary block. Do you want me to do that? It is not hard, but I didn't end up doing that because we might be adding too much optimization in this pass which should ideally be done by other passes later. Currently we run this pass at the end and don't run any other optimization passes after this, I already baked some amount of optimization into it though.

Reply:
Tried removing it without using `BranchSeeker`: https://github.com/WebAssembly/binaryen/pull/6519/commits/8c8fcc54de0053ba8c1d99d5d93a695a26ac6c3b